src/python/tritonfrontend/CMakeLists.txt

```diff
@@ -0,0 +1,182 @@
+# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+cmake_minimum_required(VERSION 3.18)
+
+# ================= Ensures Package is Structured Properly ==================
+file(COPY __init__.py DESTINATION .)
+file(COPY py.typed DESTINATION .)
+file(COPY _c/__init__.py DESTINATION ./_c/.)
+file(COPY _c/__init__.pyi DESTINATION ./_c/.)
+
+file(GLOB PYTHON_MODULE_FILES ./_api/*.py)
+file(COPY ${PYTHON_MODULE_FILES} DESTINATION ./_api/.)
+# ================================= END =====================================
+
+
+# =================== Downloading and Installing pybind11 ===================
+include(FetchContent)
+
+FetchContent_Declare(
+    pybind11
+    GIT_REPOSITORY https://github.com/pybind/pybind11.git
+    GIT_TAG v2.13.1
+    GIT_SHALLOW ON
+)
+
+FetchContent_MakeAvailable(pybind11)
+# ================================= END =====================================
+
+# ================== Collect the Dependencies ===============================
+set(
+  PYTHON_FRONTEND_BINDING_DEPS
+  ../../shared_memory_manager.h
```

Comment:
I see we actively using `../../` relative path. 
Why don't include directory instead?

Reply:
I wasn't entirely sure if it was good practice since `python` is nested underneath `src`.

Reply:
Shouldn't the better way be that the dependencies are exposed as CMake targets? I am actually surprised that the files are not implicitly included as `http/grpc-endpoint-library` are the ones having dependency

Reply:
Will follow up in DLIS-7071